src/test/java/com/google/api/codegen/testdata/python_test_library.baseline

```diff
@@ -117,6 +124,42 @@ class TestLibraryServiceClient(unittest2.TestCase):
         self.assertRaises(errors.GaxError, client.get_shelf, name, options_)
 
     @mock.patch('google.gax.config.create_stub')
+    def test_list_shelves(self, mock_create_stub):
+        # Mock gRPC layer
+        grpc_stub = mock.Mock()
+        mock_create_stub.return_value = grpc_stub
+
+        client = library_service_client.LibraryServiceClient()
+
+        # Mock response
+        next_page_token = ''
+        shelves_element = library_pb2.Shelf()
+        shelves = [shelves_element]
+        expected_response = library_pb2.ListShelvesResponse(next_page_token, shelves)
+        grpc_stub.ListShelves.return_value = expected_response
+
+        paged_list_response = client.list_shelves()
+        resources = list(paged_list_response)
+        self.assertEqual(1, len(resources))
+        self.assertEqual(expected_response.shelves[0], resources[0])
+
+        grpc_stub.ListShelves.assert_called_once()
+
+    @mock.patch('google.gax.config.API_ERRORS', (CustomException,))
+    @mock.patch('google.gax.config.create_stub')
```

Comment:
We should use `spec=` when creating `MagicMock` objects if possible.

This does *not* block merge so long as an issue is made.